core/sdk-core/src/main/java/software/amazon/awssdk/core/runtime/transform/AsyncStreamingRequestMarshaller.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.runtime.transform;
+
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+
+/**
+ * Augments a {@link Marshaller} to add contents for an async streamed request.
+ *
+ * @param <T> Type of POJO being marshalled.
+ */
+@SdkProtectedApi
+public final class AsyncStreamingRequestMarshaller<T> implements Marshaller<T> {
```

Comment:
Looks like this class is almost same as `StreamingRequestMarshaller` except the `requestBody`, can we create an abstract class to hold those common fields? It would be easier to maintain in the future if we want to add another field. We can also get rid of `StreamingMarshallerUtil` if we have the parent class